src/components/rockets/oneRocket.js

```diff
@@ -0,0 +1,71 @@
+import { useDispatch } from 'react-redux';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { bookingRocket, cancelRocket } from '../../redux/rockets/rockets';
+import './oneRocket.css';
+
+const OneRocket = ({ rockets }) => {
+  const {
+
+    id,
+    rocketName,
+    description,
+    flickrImages,
+    reserved,
+  } = rockets;
+
+  const dispatch = useDispatch();
+
+  const booking = (id) => {
+    dispatch(bookingRocket(id));
+  };
+
+  const canceling = (id) => {
+    dispatch(cancelRocket(id));
+  };
+
+  return (
+    <div className="rocket" id={id}>
+      <img src={flickrImages} alt="" />
+      <div className="column">
+        <h2 className="rocket-name">{rocketName}</h2>
+        <div className="badge">
+
+          { reserved
+            ? <button type="button" className="reserved">reserved</button>
+            : ''}
+
+          <p className="rocket-description">{description}</p>
+        </div>
+        { reserved
+          ? <button onClick={() => canceling(id)} className="cancel-button" type="button">Cancel Rocket</button>
+          : <button type="button" onClick={() => booking(id)} className="booking-button">Reserve Rocket</button>}
+      </div>
+    </div>
+  );
+};
+
+OneRocket.defaultProps = {
+  reserved: false,
+};
+
+OneRocket.defaultProps = {
+  rocketName: true,
+};
+
+OneRocket.defaultProps = {
+  description: true,
+};
+
+OneRocket.defaultProps = {
+  rockets: true,
+};
+
+OneRocket.propTypes = {
+  rocketName: PropTypes.string,
+  description: PropTypes.string,
+  rockets: PropTypes.func,
```

Comment:
- [ ] This `PropType` is not passing:

![image](https://user-images.githubusercontent.com/57421823/156240700-ca3385b6-046e-45ad-8766-20798829fa51.png)

It's expecting a function, but an object is supplied.  Kindly correct this one 💪